feat: implemented pokemon overview page

jayveeannsoriano opened the pull request:
- deleted home_page.dart
- created vm and connector for pokemon overview page state and UI
- displayed the pokemon list as a 2 grid view in pokemon overview page
- modified main.dart to display the newly added page
- created reusable widgets and string extension
- modified widget_test.dart to resolve flutter lint issues

- pokemon overview page UI
![image](https://user-images.githubusercontent.com/111727585/227434132-3da18f6c-774b-4bfa-99ee-95a5c2a235b3.png)

- flutter lint analyzer results and format
![image](https://user-images.githubusercontent.com/111727585/227434215-9414d3ee-d99f-4b44-bb5c-9ac4837fb0a9.png)

Commit:
feat: implemented pokemon overview page
- deleted home_page.dart
- created vm and connector for pokemon overview page state and UI
- displayed the pokemon list as a 2 grid view in pokemon overview page
- modified main.dart to display the newly added page
- created reusable widgets and string extension
- modified widget_test.dart to resolve flutter lint issues

Commit:
feat: implemented pokemon overview page
- deleted string_extension.dart
- added dartX package to utilize capitalize method for pokemon name string format
- modified pokemonImageUrl and manipulated string to replace index with the pokemonId

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -0,0 +1,33 @@
+    return MaterialApp(
+      home: Scaffold(
+        appBar: AppBar(
+          title: pokemonOverviewTitle,
```
no need for extra trailing comma 🙂

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -0,0 +1,33 @@
+          padding: gridViewPadding,
+          gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: gridCount),
+          itemCount: pokemons.length,
+          itemBuilder: (context, index) {
```
rename `context` to `_` since it's unused 🙂

shawn-ffufm commented on lib/utils/constants.dart:
```diff
@@ -1,5 +1,19 @@
 //TODO: added for testing GetPokemonDetailsAction. This will be removed
 const pokemonNameInput = 'jigglypuff';
+
+const pokemonOverviewTitle = Text('Pokemon Overview');
```
make it only a String not a Text widget 😅

shawn-ffufm commented on lib/utils/constants.dart:
```diff
@@ -1,5 +1,19 @@
 const pokemonNameInput = 'jigglypuff';
+
+const pokemonOverviewTitle = Text('Pokemon Overview');
+const gridViewPadding = EdgeInsets.all(10.0);
```
seems no need for this 🙂

jayveeannsoriano replied:
these values were used in pokemon_overview_page.dart
![image](https://user-images.githubusercontent.com/111727585/227459918-b695c85d-6069-4d42-bcd4-0e9c7a0f51ea.png)

shawn-ffufm replied:
yep, I meant no need to declare a constant just immediately place it there 🙂

shawn-ffufm commented on lib/utils/constants.dart:
```diff
@@ -1,5 +1,19 @@
+
+const pokemonOverviewTitle = Text('Pokemon Overview');
+const gridViewPadding = EdgeInsets.all(10.0);
+const int gridCount = 2;
```
same here 🙂

jayveeannsoriano replied:
these values were used in pokemon_overview_page.dart 
![image](https://user-images.githubusercontent.com/111727585/227460172-8595cc07-8463-49d8-a826-4f925efa927c.png)

shawn-ffufm replied:
same answer above 🙂

shawn-ffufm commented on lib/utils/constants.dart:
```diff
@@ -1,5 +1,19 @@
+const int gridCount = 2;
+const urlSplitter = '/';
+const int idIndex = 6;
+const indexUrlReplace = 'index';
```
kindly rename this to `indexKey` 🙂

shawn-ffufm commented on lib/utils/constants.dart:
```diff
@@ -1,5 +1,19 @@
+const indexUrlReplace = 'index';
+const double imageWidth = 200.0;
+const double imageHeight = 130.0;
+const double gridContentSpacing = 20.0;
```
seems no need for this 🙂

jayveeannsoriano replied:
these values were used in pokemon_tile_card.dart 
![image](https://user-images.githubusercontent.com/111727585/227460249-0c85cc22-13e2-4473-ab36-5aef6c7ca326.png)

shawn-ffufm commented on lib/utils/constants.dart:
```diff
@@ -1,5 +1,19 @@
+const double imageWidth = 200.0;
+const double imageHeight = 130.0;
+const double gridContentSpacing = 20.0;
+const pokemonImageUrlPath =
```
don't forget `[index]` since that'll be the key to be replaced 😉

shawn-ffufm commented on lib/widgets/pokemon_tile_card.dart:
```diff
@@ -0,0 +1,33 @@
+class PokemonTileCard extends StatelessWidget {
+  const PokemonTileCard({
+    Key? key,
+    required this.pokemon,
```
make it as a habit to place required properties on top of everything 🙂

shawn-ffufm commented on lib/widgets/pokemon_tile_card.dart:
```diff
@@ -0,0 +1,33 @@
+  Widget build(BuildContext context) {
+    final pokemonId = pokemon.url.split(urlSplitter)[idIndex];
+    final pokemonImageUrl = pokemonImageUrlPath.replaceAll(indexUrlReplace, pokemonId);
+    final pokemonName = pokemon.name.capitalize();
```
liked what you did above, it's neat 💪

shawn-ffufm commented on lib/widgets/spacing.dart:
```diff
@@ -0,0 +1,15 @@
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(height: spacing);
```
just use fat arrow notation 🙂

shawn-ffufm reviewed (changes requested):
see pr feedback 🙂

Commit:
feat: implemented pokemon overview page
- removed trailing commas
- renamed context to _ in pokemon_overview_page.dart
- made pokemonOverviewTitle as const String instead of a Text widget
- renamed indexUrlReplace to indexKey
- modified indexKey value to [index]
- modified pokemonImageUrlPath from index to [index]
- moved required properties on top
- replaced bracket to a fat arrow notation in spacing.dart

Commit:
feat: implemented pokemon overview page
- removed declared constants and placed values direclty in widget properties
